src/Credit/FileUtils.hpp

```diff
@@ -30,12 +30,10 @@ std::unique_ptr<char> getChosenFilePath() {
         filename += ".txt";
     }
 
-    std::unique_ptr<char> pathC;
```

Comment:
I'd prefer to stick with smart pointers for this type of thing, if possible. Here's a possible solution, though I'm sure there are ways to improve on it:
- We define a type `using path_ptr = std::unique_ptr<char, decltype(&std::free)>;` (along with a comment describing why the type exists).
- We create `path_ptr pathC { {}, std::free };`.
- The function here returns a `path_ptr` instead of a `char` pointer.

Does that seem okay to you?

Reply:
yes, I guess.. seems a bit overcomplex tbh. rack has a nice macro called `DEFER` that is used for these types of things, we can put some code inside it and it runs at the end of the function scope.

would prefer for you to try this part, I am not familiar with smart pointer override stuff yet.